tap_five9/client.py

```diff
@@ -48,23 +50,41 @@ def run_report(self, params):
 
         LOGGER.exception('error running five9 report')
 
-    def get_report_results(self, identifier):
+    def get_report_results(self, identifier, config=None):
+        LOGGER.info(f'five9 report created with identifier: {identifier}')
 
-        for num_retries in range(self.MAX_RETRIES):
-            LOGGER.info(f'five9 getting report results')
-            try:
-                response = self.client.configuration.getReportResult(identifier)
-            except zeep.exceptions.Fault:
-                LOGGER.info('five9 result not ready')
-                time.sleep(15)
-                continue
-
-            if response:
-                fields = [self.inflect_field(field) for field in response.header['values']['data']]
-                return self.client.parse_response(fields, response.records)
+        poll_timeout = self.POLL_TIMEOUT
+        poll_delay = self.POLL_DELAY
+        if config.get('poll_settings'):
+            settings = config.get('poll_settings', {})
+            poll_timeout = settings.get('timeout')
+            poll_delay = settings.get('delay')
+
+        is_running = True
+        start_time = datetime.datetime.utcnow()
+        elapsed_time = 0
+        while is_running:
+            if elapsed_time > poll_timeout:
+                # Abort if the timeout occurred and return so we continue to the next stream
+                LOGGER.exception(f'five9 timed out while running the report (max {poll_timeout} sec)')
+                return []
```

Comment:
Do we want to do any retrying here to preserve previous logic? I think for now this is fine and if we see that it fails continuously, then we can revisit this.

Reply:
makes sense, will keep an eye out for this error message while monitoring the tap dashboard after deploying